ldms/src/sampler/perfevent.c

```diff
@@ -511,13 +519,21 @@ static int sample(struct ldmsd_sampler *self)
 	}
 
 	base_sample_begin(base);
-
+	static int readerrlogged = 0;
 	struct event_group *eg;
 	LIST_FOREACH(eg, &gevent_list, entry) {
 		unsigned int read_size = (eg->eventCounter + 2) * sizeof(long long); /*based on read format. */
 		long long *data = calloc(eg->eventCounter + 2, sizeof(long long)); /*allocate memory based on read format. */
 		int read_result = read(eg->leader, data, read_size); /* do the read */
-		int event_index = (read_result / sizeof(long long)) - 1; /* start from the last event added to the list */
+		if (read_result < 0) {
+			free(data);
+			if (!readerrlogged) {
+				msglog(LDMSD_LERROR, "perfevent: read event failed.\n");
+				readerrlogged = 1;
+			}
+			break;
+		}
+		/* int event_index = (read_result / sizeof(long long)) - 1;  start from the last event added to the list? */
```

Comment:
not sure why this is left here in a comment. are you not certain of this code? has it been tested? are any of these changes symptomatic?

Reply:
left it just in case someone recognized it as desired debugging code. Six weeks without comment seems to indicate it is not, so I removed it just now.